Authentication and Admin Service

yhtMinceraft1010X opened the pull request:
Let's use Firebase to manage the user identities and admin roles.

For GitHub Auth, logging in to a new user account (CREATE) and 
logging in to an already existing account can be done in the 
same step.

yhtMinceraft1010X committed:
Transfer admin service files

yhtMinceraft1010X committed:
Delete unused nodemon file

yhtMinceraft1010X committed:
Add frontend files

yhtMinceraft1010X committed:
Transfer main page Firebase integrations

yhtMinceraft1010X committed:
Transfer README lines

yhtMinceraft1010X committed:
Add firebase tools package

yhtMinceraft1010X committed:
Add system tests to admin service and rename a unit test file

yhtMinceraft1010X committed:
Edit path for GET list all users for clarity

yhtMinceraft1010X committed:
Refactor passing in of env variables for admin service

yhtMinceraft1010X committed:
Remove comments about connecting directly to user service

yhtMinceraft1010X committed:
Add number typing to port

ong6 commented on frontend/src/firebase-client/firebase_config_dev.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "apiKey": "AIzaSyBqF7f50uQXoGOU9c9oGfwmt6vtxGotyr0",
+  "authDomain": "peerprep-group11-dev.firebaseapp.com",
+  "projectId": "peerprep-group11-dev",
+  "storageBucket": "peerprep-group11-dev.appspot.com",
+  "messagingSenderId": "912506881405",
+  "appId": "1:912506881405:web:dca2a2d1096dac14267814"
+}
```
are these keys we can expose?

yhtMinceraft1010X replied:
I think these should be safe and needed for the client to know which app to talk to for the authentication

ong6 replied:
ok lgtm, feel free to fix conficts and merge

ong6 reviewed (approved):
LGTM, thanks for adding test cases too

yhtMinceraft1010X committed:
Merge branch 'master' into admin-service-branch

yhtMinceraft1010X committed:
Rename api to routes in admin-service

yhtMinceraft1010X committed:
Merge branch 'master' into admin-service-branch

yhtMinceraft1010X committed:
Remove gateway links in frontend

Since gateway is not yet merged, errors may occur here.

Let's remove the code for gateway connection.
